streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1148,6 +1148,17 @@ private static Pattern buildPattern(final Collection<String> sourceTopics,
         return results;
     }
 
+    public List<String> sourceTopicsForStore(final String storeName) {
```

Comment:
a side note: Should we also consider the map `stateStoreNameToSourceRegex` here?

Reply:
I am not sure, do you think it will identify more topic groups? At the end of the day we are identifying a topic groupId to which a store belongs to. 

On a side note, if we do want to add stateStoreNameToSourceRegex along with stateStoreNameToSourceTopics, then we can probably return a set instead of list in this function and iterate over both of above one by one.